core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -87,17 +87,23 @@ import scala.util.control.ControlThrowable
  * @param logDirs The directories where offset checkpoints reside
  * @param logs The pool of logs
  * @param time A way to control the passage of time
+ * @param scheduler The thread pool scheduler used for background actions
  */
 class LogCleaner(initialConfig: CleanerConfig,
                  val logDirs: Seq[File],
                  val logs: Pool[TopicPartition, UnifiedLog],
                  val logDirFailureChannel: LogDirFailureChannel,
-                 time: Time = Time.SYSTEM) extends Logging with KafkaMetricsGroup with BrokerReconfigurable
+                 time: Time = Time.SYSTEM,
+                 private[log] val scheduler: Scheduler) extends Logging with KafkaMetricsGroup with BrokerReconfigurable
 {
 
   /* Log cleaner configuration which may be dynamically updated */
   @volatile private var config = initialConfig
 
+  // Visible for testing
+  val housekeepingDelayMs = 30000
+  val housekeepingIntervalMs = 30000
```

Comment:
30 seconds seems quite frequent. Could we do 5 mins?

Reply:
Updated.